runner/runner.py

```diff
@@ -16,46 +17,32 @@
 obligatory_plugins = [PipelineAnalyser(), IntegrityChecker()]
 
 
-def overwrite_model_configs(config: Experiment, pipeline: Pipeline) -> Pipeline:
+def overwrite_model_configs_(config: Experiment, pipeline: Pipeline) -> None:
     for key, value in vars(config).items():
         if value is not None:
             for model in flatten(pipeline.children()):
                 if hasattr(model, "config"):
                     if hasattr(model.config, key):
                         vars(model.config)[key] = value
 
-    return pipeline
 
-
-def add_position_to_block_names(pipeline: Pipeline) -> Pipeline:
-    entire_pipeline = pipeline.children()
-
-    def add_position(block: Union[List[Element], Element], position: int, prefix: str):
-        if isinstance(block, List):
-            if position > 0:
-                prefix += f"{position - 1}-"
-            for i, child in enumerate(block):
-                add_position(child, i, prefix)
-        elif not isinstance(block, DataSource):
-            block.id += f"{prefix}{position}"
-
-    add_position(entire_pipeline, 1, "-")
-
-    return pipeline
-
-
-def append_pipeline_id(pipeline: Pipeline) -> Pipeline:
+def append_parent_path_and_id_(pipeline: Pipeline) -> None:
     entire_pipeline = pipeline.dict_children()
 
-    def append_id(block, pipeline_id: str):
-        block["obj"].pipeline_id = f"{pipeline_id}"
+    def append(block, parent_path: str, id_with_prefix: str):
+        block["obj"].parent_path = parent_path
+        if not isinstance(block["obj"], DataSource):
+            block["obj"].id += id_with_prefix
```

Comment:
do we need this?

Reply:
I don't understand your question 😄 
This does two things
- Correctly places the path of the model (so we can save it correctly in a directory)
- Adds a uniquely ``id`` on each model so we can ``load`` & ``save`` the correct one.

Reply:
yes, sorry, all good! just had a thought that we still treat Datasource separately, but we do - we don' want to append anything to it. great! let's merge it, thank you!